analyzer/codechecker_analyzer/cmd/analyze.py

```diff
@@ -559,6 +559,28 @@ def __update_skip_file(args):
         shutil.copyfile(args.skipfile, skip_file_to_send)
 
 
+def __cleanup_metadata(metadata_prev, metadata):
+    """ Cleanup metadata.
+
+    If the source file for a plist does not exists, remove the plist from the
+    system and from the metadata.
+    """
+    if not metadata_prev:
+        return
+
+    result_src_files = metadata_prev['result_source_files']
+    for plist_file, source_file in result_src_files.items():
+        if not os.path.exists(source_file):
+            try:
+                LOG.info("Remove plist file '%s' because it refers to a "
+                         "source file ('%s') which was removed.",
+                         plist_file, source_file)
+                del metadata['result_source_files'][plist_file]
+                os.remove(plist_file)
+            except (OSError, KeyError):
```

Comment:
As the iteration is on the collection, plist_file key must exist. Hence KeyError is not possible here. If only OSError remains, then it would be easier to see, that only filesystem access can go wrong.